main.py

```diff
@@ -14,98 +22,70 @@ def __init__(self):
     def hash_password(self, password):
         """Hashes the password using SHA-256."""
         hashed_password = hashlib.sha256(password.encode()).hexdigest()
-        return password
-    
+        return hashed_password
+
     def verify_username(self, username):
-        good = "good"
-        if username < 1:
-            good = "bad"
-        else:
-            good = "good"
-            if username > 30:
-                good = "bad"
-            else:
-                good = "good"
-        return good
-                
-        
-    
+        if 1 < len(username) < 30:
+            return True
+        log.info("username must be between 1 and 30 characters.")
+        return False
+
     def verify_email(self, email):
-        good = "good"
-        if "@" in email:
-            good = "good"
-            if "." not in email:
-                good = "bad"
-            else:
-                good = "good"
-        else:
-            good = "bad"
-        return good
-        
-        
+        if re.fullmatch(r"[^@]+@[^@]+\.[^@]+", email):
+            return True
+        log.info("email did not contain '@' and/or '.'")
+        return False
+
     def verify_password(self, password, email):
-        good = "good"
         weak_passwords = ["password", "123456", "qwerty", "abc123"]
+
         if len(password) < 8:
-            good = "bad"
-        else:
-            good = "good"
-            if password.lower() in weak_passwords:
-                good = "bad"
-            else:
-                good = "good"
-                if email == password.lower():
-                    good = "bad"
-                else:
-                    good = "good"
-                    has_lower = False
-                    has_upper = False
-                    has_digit = False
-                    has_special = False
-                    for c in password:
-                        if c.islower():
-                            has_lower = True
-                        elif c.isupper():
-                            has_upper = True
-                        elif c.isdigit():
-                            has_digit = True
-                        elif c in "!@#$%^&*()_+":
-                            has_special = True
-                    if has_lower == False:
-                        good = "bad"
-                    if has_upper == False:
-                        good = "bad"
-                    if has_digit == False:
-                        good = "bad"
-                    if has_special == False:
-                        good = "bad"
-        return good
-        
+            log.info("Password must be at least 8 characters long.")
+            return False
+
+        if password.lower() in weak_passwords:
+            log.info("Password is too weak.")
+            return False
```

Comment:
```suggestion
```

3 of the passwords in weak_passwords won't even reach this check because they aren't 8+ characters long. In my opinion, just remove this check and the weak_passwords list. If the password passes all the other `if` checks, it won't be a weak password anyway.